tiledb/sm/serialization/query.cc

```diff
@@ -261,84 +261,122 @@ Status subarray_to_capnp(
   return Status::Ok();
 }
 
-Status subarray_from_capnp(
-    const capnp::Subarray::Reader& reader, Subarray* subarray) {
-  RETURN_NOT_OK(subarray->set_coalesce_ranges(reader.getCoalesceRanges()));
+Subarray subarray_from_capnp(
+    const capnp::Subarray::Reader& reader,
+    const Array* array,
+    Layout layout,
```

Comment:
It looks like there was a preexisting defect that the layout was being written in `subarray_to_capnp` but not read back in this function. It shouldn't be an external argument, certainly.